lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -392,6 +400,81 @@ sub _in_container {
     return -r '/.dockerenv';
 }
 
+=head2 _linux_flock_data
+
+Param: lock type. It can be F_WRLCK or F_UNLCK
+
+return: A FLOCK structure
+
+=cut
+
+# The following code is from https://docstore.mik.ua/orelly/perl4/cook/ch07_26.htm
+sub _linux_flock_data {
+    my ($type) = @_;
+    my $FLOCK_STRUCT = "s s l l i";
+    return pack($FLOCK_STRUCT, $type, SEEK_SET, 0, 0, 0);
+}
```

Comment:
Do we really need region locks like this?  We're typically looking at areas near or at EOF, so wouldn't a simpler mutex-like locking be OK?

Reply:
We needn't region lock here. So that's why I used `0,0` here . We don't care about the region. But, if you are talking about `perldoc -f flock`,  We have a discussion here: https://deriv-group.slack.com/archives/C87ASKE79/p1625146660393300

Reply:
Let me copy the discussion here for clarify: 

 Context: I want to use flock to lock the log file, but that log file is opened before fork, so the FD is inherited by 2 subprocesses, so every  subprocess can get flock even another subprocess flock ed it already. that make the lock useless. 

.... 've been looking around just now maybe you need to use fcntl F_SETLKW instead of flock because it's based on the process id not the fd: